core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -611,15 +620,31 @@ private[spark] class TaskSetManager(
 
     // If no executors have registered yet, don't abort the stage, just wait.  We probably
     // got here because a task set was added before the executors registered.
-    if (executors.nonEmpty) {
+    if (executorsByHost.nonEmpty) {
       // take any task that needs to be scheduled, and see if we can find some executor it *could*
```

Comment:
Can you move this comment up to the beginning of the method (above where pendingTask is defined)